AsyncDisplayKit/ASMultiplexImageNode.mm

```diff
@@ -478,6 +450,37 @@ - (UIImage *)_bestImmediatelyAvailableImageFromDataSource:(id *)imageIdentifierO
 }
 
 #pragma mark -
+
+- (void)_updateProgressImageBlockOnDownloaderIfNeeded
+{
+    // Read our interface state so that we don't lock super while holding our lock.
+    ASInterfaceState interfaceState = self.interfaceState;
```

Comment:
If this is called while the _downloadIdentifierLock is held, won't that result in the issue you're commenting about? It's called with the lock held in visibilityDidChange:?